tests/console/nvme_checks.pm

```diff
@@ -0,0 +1,116 @@
+# SUSE's openQA tests
+#
+# Copyright © 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved. This file is offered as-is,
+# without any warranty.
+
+# Summary: Smoke tests for NVM Express Controller.
+#
+# - Install nvme-cli which is a tool to manage NVM Express
+# - Check sys path that should be hold details
+# - Check if we have any NVMe controllers on the PCI bus
+# - Check if we have NVMe devices
+# - Check if we have namespace
+# - Send Identify Admin Command to the NVMe controller
+# - Issue a Read command both using the helper and manually
+# - Compare it to what we have with a regular dd
+#
+# Maintainer: QA SLE YaST team <qa-sle-yast@suse.de>
+
+use base "installbasetest";
+use strict;
+use warnings;
+use testapi;
+use utils;
+use scheduler 'get_test_suite_data';
+
+sub _smoke_checks {
+    my $nvm_test_data = shift;
```

Comment:
i want to make sure that it clear what it should expect. the subroutine can be extended later. it will be always test_data but not necessary disk data

Reply:
if you call the argument in the same way in every single method should be fine. 

For extension, my other point is that you are not preparing your test data in case you have sub-section of your test data, that at first glance it looks like that when you use that naming (meaning I was expecting some parent nodes for L2-11 in your test data) like you were passing a subset or something like that. But anyway if you apply same name for now should be fine.